src/engine/Map.cxx

```diff
@@ -394,34 +394,52 @@ Point Map::findNodeInMap(const SDL_Point &screenCoordinates) const
   return foundCoordinates;
 }
 
-void Map::demolishNode(const Point &isoCoordinates, bool updateNeighboringTiles)
+void Map::demolishNode(const std::vector<Point> &isoCoordinates, bool updateNeighboringTiles, Layer layer)
 {
-  const size_t index = isoCoordinates.x * m_columns + isoCoordinates.y;
-  if (index < mapNodes.size())
+  std::vector<Point> nodesToDemolish;
+  for (auto &isoCoord : isoCoordinates)
   {
-    const Layer layer = Layer::BUILDINGS;
-    const Point origCornerPoint = mapNodes[isoCoordinates.x * m_columns + isoCoordinates.y]->getOrigCornerPoint(layer);
-    const size_t origIndex = origCornerPoint.x * m_columns + origCornerPoint.y;
-
-    if (origIndex < mapNodes.size() && mapNodes[origIndex])
+    MapNode *node = mapNodes[isoCoord.x * m_columns + isoCoord.y].get();
+    if (isPointWithinMapBoundaries(isoCoord))
     {
-      const std::string &tileID = mapNodes[origIndex]->getTileID(layer);
-      std::vector<Point> objectCoordinates = getObjectCoords(origCornerPoint, tileID);
+      // Check if there's something on the BUILDINGS layer before checking origCornerPoint
+      if (node->getMapNodeDataForLayer(Layer::BUILDINGS).tileData)
```

Comment:
I think we should use layer here instead of BUILDINGS?

Reply:
for now, multiobjects are only placed on BUILDINGS layer. when we call demolish node for another layer, let's say GROUNDDECORATION, we also need to gather all nodes that are beneath the multinode object, even if we only touch one point, and this is on the BUILDINGS layer and not on the GROUNDDECORATION layer.
I've added this as comment too